### ISSUE TITLE
Custom Login page

### ISSUE THREAD

**Issue by cristianomariacumer** [CatalystIT-AU/moodle-auth_saml2#20]
Hi @brendanheywood ,

I have implemented a custom login page, like the to one moodle uses for CAS authentication.
I have done that because we are using both IdP users and local moodle users and I find that unskilled users are confused by the standard login page:

![screen shot 2016-07-14 at 17 27 16](https://cloud.githubusercontent.com/assets/19989683/16845171/4408cf74-49e8-11e6-91b9-c1879f23bdd2.png)

I'm quit sure that in this case 99% of my users would punch their credentials directly in the login/password box instead of using our idp (ScientificNet).

So I added this intermediate page:

![screen shot 2016-07-14 at 17 29 07](https://cloud.githubusercontent.com/assets/19989683/16845233/81c3e628-49e8-11e6-9f99-5947e1910740.png)

Where it's possible to explain the different possibilities.

Let me know if you are interested in this code.

Cristiano

**Comment by brendanheywood** [comment]
hi @cristianomariacumer,

I'm am interested in this feature, but I'm leaning towards it actually being a core moodle feature and not in here. This logic really should be part of the core login page and apply to any moodle auth plugin that declares remote login pages via loginpage_idp_list(). I'll make a tracker and see whether HQ are interested or not.

See:

https://tracker.moodle.org/browse/MDL-55257

**Comment by xow** [comment]
I agree that it would be good to have this option in core. Plus we can look more holistically at the experience of having multiple identity providers or having one that is more commonly used than manual auth. Any patches contributed to https://tracker.moodle.org/browse/MDL-55257 are appreciated!

**Comment by JorgenIvarsson** [comment]
Hi Cristiano! How do I get your code? I seems to have the same problem, switching from CAS to SAML2.

**Comment by cristianocumer** [comment]
Hi @JorgenIvarsson ,

still interested? Sorry, I didn't notice your comment

Cristiano

**Comment by brendanheywood** [comment]
fyi there has been a bit of movement in the core tracker re https://tracker.moodle.org/browse/MDL-55257
